server/server_test.go

```diff
@@ -47,131 +58,85 @@ func TestServerStart(t *testing.T) {
+// 	var errStop error
+// 	go func() {
+// 		close(serviceRunning)
+// 		err = Start("0.0.0.0:5567")
 
-	if response.Code != http.StatusOK {
-		t.Fatalf("Something wrong")
-	}
-}
+// 		defer close(serviceDone)
+// 	}()
 
-func TestServerPutHandlerNotFound(t *testing.T) {
-	var testServer = NewServer()
-	err := testServer.Put("/test", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		t.Log("PUT")
-	})
+// 	<-serviceRunning
 
-	if err != nil {
-		t.Errorf("Error Occurred: %v", err)
-	}
+// 	if err != nil {
+// 		t.Fatalf("Server never started %v", err)
+// 	}
+// 	errStop = Stop()
+// 	if errStop != nil {
+// 		t.Fatalf("Server never started %v", errStop)
+// 	}
 
-	request, _ := http.NewRequest(http.MethodPut, "/testFuncNotFound", nil)
-	response := httptest.NewRecorder()
+// 	<-serviceDone
+// }
 
-	testServer.ServeHTTP(response, request)
+func TestServerGetConfiguration(t *testing.T) {
+	config := GetConfiguration()
 
-	if response.Code == http.StatusOK {
+	if config.ReadTimeoutInSeconds != DefaultWriteTimeOutInSeconds {
 		t.Fatalf("Something wrong")
 	}
 }
 
-func TestServerPutSubHandlerNotFound(t *testing.T) {
-	var testServer = NewServer()
-	err := testServer.Put("/mufasa", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		t.Log("PUT")
-	})
-
-	if err != nil {
-		t.Errorf("Error Occurred: %v", err)
-	}
-
-	request, _ := http.NewRequest(http.MethodPut, "/testFuncNotFound/Fail", nil)
-	response := httptest.NewRecorder()
-
-	testServer.ServeHTTP(response, request)
-
-	if response.Code == http.StatusOK {
+func TestServerGetLogger(t *testing.T) {
+	logger := GetLogger()
+	if logger == nil {
 		t.Fatalf("Something wrong")
 	}
 }
 
-func TestServerPutMethodNotFound(t *testing.T) {
+func TestServerFuncAddInvalidRoute(t *testing.T) {
 	var testServer = NewServer()
-	err := testServer.Put("/test", func(w http.ResponseWriter, r *http.Request) {
+	err := testServer.GetFunc("[ INVALID REGEX", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-		t.Log("PUT")
 	})
 
-	if err != nil {
-		t.Errorf("Error Occurred: %v", err)
-	}
-
-	request, _ := http.NewRequest(http.MethodGet, "/test", nil)
-	response := httptest.NewRecorder()
-
-	testServer.ServeHTTP(response, request)
-
-	if response.Code == http.StatusOK {
-		t.Fatalf("Something wrong")
+	if err == nil {
+		t.Error("This should have failed an an invalid regex was passed int to GET")
 	}
 }
 
 func TestServerAddInvalidRoute(t *testing.T) {
 	var testServer = NewServer()
-	err := testServer.Get("[ INVALID REGEX", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	})
+	err := testServer.Get("[ INVALID REGEX", &HrpleTestHandler{})
 
 	if err == nil {
 		t.Error("This should have failed an an invalid regex was passed int to GET")
 	}
 }
+
+type HrpleTestHandler struct {
+}
+
+func (h *HrpleTestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [parameter 'r' seems to be unused, consider removing or renaming it as _](https://app.codacy.com/gh/hrple/common/pullRequest?prid=5576416)